### PR TITLE
Add view color feature

### PR DESCRIPTION
**Pull request by sumanthratna**
Implements #2119. 

NOTE: the types are all wrong, I'd appreciate it if someone else fixes the types (but the Chrome extension still works).

### CODE REVIEW

File: source/features/view-color.tsx [hunk 1]
```diff
@@ -0,0 +1,166 @@
+import features from '../libs/features';
+
+function init(): void {
+	addColors();
+}
+const namedColors: string = [
```
Comment (jerone) [inline source/features/view-color.tsx]: Can probably be replaced by some npm package, like [color-name-list](https://www.npmjs.com/package/color-name-list)...

**Comment by jerone** [comment]
If you copy the whole source of the [mentioned UserScript](https://github.com/sindresorhus/refined-github/issues/2119#issuecomment-498982592), you probably have to ask permission from the author.

**Comment by sumanthratna** [comment]
> If you copy the whole source of the [mentioned UserScript](https://github.com/sindresorhus/refined-github/issues/2119#issuecomment-498982592), you probably have to ask permission from the author.

He's okay with it!

**Comment by bfred-it** [comment]
I did ask to integrate his user scripts, but instead of copy-pasting the code and then having to support it separately, I’d rather `require` the script, add the globals needed (`GM.addStyle` and the events we need) and build a regular feature around it. 

This would enable us to just follow any updates Mottie has in the future, rather than forking his code, and add more features userscripts the same way in the future.

To do so, we need to keep the feature as two files:

- `color-preview.user.js`: 99% copy-pasted file, excluded from XO, only changes to export a function and disable the listeners
- `color-preview.tsx`: the actual feature wrapper that has the listeners and feature initialization. It will import the user.js file and run it as necessary.

File: source/features/view-color.tsx [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+			if (!status.done) {
+				requestAnimationFrame(loop);
+			}
+		};
+		loop();
+	}
+}
+features.add({
+	id: 'view-color',
+	description: 'View various color codes',
```
Comment (bfred-it) [inline source/features/view-color.tsx]: “All color previews next to colors everywhere”

**Comment by jerone** [comment]
> I did ask to integrate his user scripts, but instead of forking the code and then having to support it separately, I’d rather `require` the script, add the globals needed (`GM.addStyle` and the events we need) and build a regular feature around it.

Interesting idea. That will open up using other UserScripts too.



> Alternatively we can probably `npm install` and require the file directly from GitHub and then use the existing webpack replace loader do make these changes on the fly. So no copy-pasting required.

That will work:
```cli
npm install --save git+https://github.com/Mottie/GitHub-userscripts.git#v1.0.241
```

**Comment by bfred-it** [comment]
npm natively supports installing from GitHub using `npm install user/repo`

**Comment by sumanthratna** [comment]
I think this PR should just be reimplemented from scratch, using the suggestions mentioned above.

**Comment by fregante** [comment]
> I did ask to integrate his user scripts,

Change of direction, for posterity: **if a userscript/extension already exist, just use that.**

Refined GitHub does not have to include everything, we already have a lot of stuff to take care of.

**Pull request closed without merging**